### DISCUSSION TITLE
Different error pages depending on error code

### DISCUSSION THREAD

**Discussion by YassinS** [pimcore/pimcore#17122] [Q&A / Support]
Hello,

I am trying to display either a 404 not found error page or a 500 error page with pimcore. So far we have been using a generic error page via the system settings dialog like so:
![Screenshot 2024-06-04 at 10 35 07](https://github.com/pimcore/pimcore/assets/11740576/6786e0c4-edda-4442-8215-faff66aacddf)
This error page gets served for each error that the user might encounter.

This works for now, but is rather confusing to users and ambiguous. I was wondering if there is a way to conditionally render an error document depending on the error code. Is there maybe a way to pass the error code to a twig template?

Thank you in advance for any ideas! :)

**Comment by alasdair-shields** [comment]
Not sure if this can be done via Pimcore admin but you could maybe accomplish it at the Symfony level:

https://symfony.com/doc/current/controller/error_pages.html

**Reply by YassinS** [reply]
Thank you, I guess that's worth a try. It would've been smoother to accomplish those error pages "the pimcore way", but if it works it works.

Thanks for your input :)

**Comment by blankse** [comment]
@YassinS You can read the status code via the controller and pass it on to the template.
```
<?php

namespace App\Controller\Document\Page;

use Pimcore\Controller\FrontendController;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;

class ErrorController extends FrontendController
{
    public function error(Request $request): Response
    {
        $statusCode = 500;
        $exception = $request->attributes->get('exception');
        if ($exception instanceof HttpExceptionInterface) {
            $statusCode = $exception->getStatusCode();
        }

        return $this->render('error.html.twig', [
            'statusCode' => $statusCode,
        ]);
    }
}
```

**Reply by YassinS** [reply]
Thank you, that's very helpful :)

**Reply by YassinS** [reply]
@blankse wait how do I get Pimcore to send all exceptions to the Controller you have provided? I tried registering it in the sevices.yaml, but without any success. I have also tried adding the Controller to the framework.yaml as error_controller.

**Reply by blankse** [reply]
@YassinS You create a Document type:
![image](https://github.com/pimcore/pimcore/assets/998558/1502b673-b1d4-4c85-8ec8-cc7fceba612c)

![image](https://github.com/pimcore/pimcore/assets/998558/45edbbae-8731-4332-90cc-cd7def500a04)

Than you can create a Page with "Exclude from Navigation".

This page you use as error page.
